vermouth/molecule.py

```diff
@@ -642,6 +661,69 @@ def share_moltype_with(self, other):
         # the interactions.
         return nx.is_isomorphic(self, other)
 
+    # TODO: Allow comparison of interactions betweem isomorphic molecules.
+    def same_interactions(self, other):
+        """
+        Returns `True` if the interactions are the same.
+
+        To be equal, two interations must share the same node key references,
+        the same interation parameters, and the same meta attributes.
+
+        Parameters
+        ----------
+        other: Molecule
+
+        Returns
+        -------
+        bool
+        """
+        return self.interactions == other.interactions
+
+    # TODO: Allow default values for attributes.
+    # In most cases, we assume that an unspecified attribute is equivalent to
+    # the attribute being None; we should allow for this in the comparison. We
+    # should also allow an attribute to have an arbitrary default value. For
+    # instance, the assumed default for the `PTM_atom` attribute is False.
+    def same_nodes(self, other, ignore_attr=()):
+        """
+        Returns `True` if the nodes are the same and in the same order.
+
+        The equality criteria used for the attribute values are those of
+        :func:`vermouth.utils.are_different`.
+
+        Parameters
+        ----------
+        other: Molecule
+        ignore_attr: collections.abc.Sequence
```

Comment:
Iterable, not Sequence

Reply:
Actually `Container`, `Iterable` does not have `__contains__` which is the only method I need here.